Python/suggestions.c

```diff
@@ -43,7 +43,7 @@ levenshtein_distance(const char *a, size_t a_size,
                      const char *b, size_t b_size,
                      size_t max_cost)
 {
-    static size_t buffer[MAX_STRING_SIZE];
```

Comment:
@pablogsal Is this static because we are trying not to allocate a big buffer on stack or some other reason? If it's the former I can change it to be dynamically allocated.

Reply:
See: https://github.com/python/cpython/issues/88215#issuecomment-1093914436

Reply:
But since we require C11 compiler, now the situation looks change.

Reply:
I also would prefer not to have a heap allocation here for perf reasons but I am happy to discuss if we really need it. The issue linked by @corona10 can give you context.

Reply:
But msvc does not supports vla so we cannot use it. My understanding is that this code is executed only if the error is not handled by the program and the interpreter is about to exit, it this correct? If so I don't think think perf matters here.

Reply:
Your reasoning is correct although we still may use in the future for other stuff, that's why is relatively generic. In any case, I am fine adding a heap allocation if we really need to 👍

Reply:
If heap allocation is preferred, it could be done in the calling function rather than at every comparison.

Reply:
That would make sense but I fear we are complicating this API quite a lot...

Reply:
Changed it to use heap allocation.

Reply:
@pablogsal @sweeneyde PTAL, thanks

Reply:
ping @pablogsal